### ISSUE TITLE
Unnecssary parenthesis lint

### ISSUE THREAD

**Issue by bilelmoussaoui** [rust-lang/rust-clippy#10557]
### What it does

Currently, if the code contains unnecessary parenthesis, clippy doesn't warn the user about it.  It would be nice if there was some new lint to detect these cases and warn the user about them. 

### Category

style

### Advantage

- Remove unnecessary parenthesis in the code

### Example

```rust
fn new_rect(x: f64, y: f64, width: f64, height: f64) {
    println!("I am a rectangle ({x},{y}) ({width}, {height})");
}

fn main() {
    let x = 0f32;
    let y = 0f32;
    let width = 100f32;
    let height = 100f32;

    new_rect((x) as f64, (y) as f64, (width) as f64, (height) as f64);
}

```

Could be written as:

```rust
fn new_rect(x: f64, y: f64, width: f64, height: f64) {
    println!("I am a rectangle ({x},{y}) ({width}, {height})");
}

fn main() {
    let x = 0f32;
    let y = 0f32;
    let width = 100f32;
    let height = 100f32;

    new_rect(x as f64, y as f64, width as f64, height as f64);
}

```

**Comment by asquared31415** [comment]
rustc has an `unused_parens` lint, this might be a shortcoming of that lint.  notably `some_fn((x))` and `let _ = (x);` *do* have a warning triggered, so it might be something to do with being part of the `as` cast there?
